src/features/game/events/landExpansion/tradeTentacle.ts

```diff
@@ -0,0 +1,76 @@
+import cloneDeep from "lodash.clonedeep";
+import Decimal from "decimal.js-light";
+import { GameState } from "features/game/types/game";
+import { getSeasonalTicket } from "features/game/types/seasons";
+import { trackActivity } from "features/game/types/bumpkinActivity";
+
+export type TradeTentacleAction = {
+  type: "shelly.tradeTentacle";
+};
+
+type Options = {
+  state: Readonly<GameState>;
+  action: TradeTentacleAction;
+  createdAt?: number;
+};
+
+export function tradeTentacle({
+  state,
+  createdAt = Date.now(),
+}: Options): GameState {
+  const stateCopy = cloneDeep(state);
+  const { bumpkin } = stateCopy;
+
+  if (!bumpkin) {
+    throw new Error("You do not have a Bumpkin");
+  }
+
+  if (!stateCopy.npcs) {
+    throw new Error("NPCs does not exist");
+  }
+
+  // insuficient tentacles
+  if (!stateCopy.inventory["Kraken Tentacle"]) {
+    throw new Error("Insufficient quantity to trade");
+  }
+
+  if (!stateCopy.npcs?.shelly) {
+    stateCopy.npcs.shelly = { deliveryCount: 0 };
+  }
+
+  const currentScales =
+    stateCopy.inventory[getSeasonalTicket()] || new Decimal(0);
+
+  const currentTentaclesInInventory =
+    stateCopy.inventory["Kraken Tentacle"] || new Decimal(0);
+
+  const currentTentaclesPlaced =
```

Comment:
Consider using `getChestItems` to see whether the item is available in chest or already placed